File: Arrays_&_Hashing/Products_of_Array_Except_Self/1_BruteForce.cpp
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int n = nums.size();
        vector<int> ans(n, 1);
        for (int i = 0; i < n; ++i) {
            long long prod = 1;
            for (int j = 0; j < n; ++j) {
                if (i == j) continue;
                prod *= nums[j];
            }
            ans[i] = static_cast<int>(prod);
        }
        return ans;
    }
};

File: Arrays_&_Hashing/Products_of_Array_Except_Self/1_BruteForce.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int n = nums.length;
        int[] ans = new int[n];
        for (int i = 0; i < n; i++) {
            long prod = 1;
            for (int j = 0; j < n; j++) {
                if (i == j) continue;
                prod *= nums[j];
            }
            ans[i] = (int) prod;
        }
        return ans;
    }
}
